pkg/tfshim/sdk-v2/provider2.go

```diff
@@ -798,3 +803,83 @@ func recoverDiagnostic(d tfprotov5.Diagnostic) diag.Diagnostic {
 	}
 	return dd
 }
+
+func normalizeBlockCollections(val cty.Value, res *schema.Resource) cty.Value {
```

Comment:
Can this tolerate val==null (I'm not sure if this ever arises, but it may during Create, worth checking). I don't think it needs to handle val==unknown, but can assert about it.